GitOut/Features/Wpf/AttachedProperties/TextZoom.cs

```diff
@@ -0,0 +1,115 @@
+        public static KeyGesture? GetZoomResetKeyGesture(DependencyObject obj) => (KeyGesture?)obj.GetValue(ZoomResetKeyGestureProperty);
+        public static void SetZoomInKeyGesture(DependencyObject obj, KeyGesture? value) => obj.SetValue(ZoomInKeyGestureProperty, value);
+        public static void SetZoomOutKeyGesture(DependencyObject obj, KeyGesture? value) => obj.SetValue(ZoomOutKeyGestureProperty, value);
+        public static void SetZoomResetKeyGesture(DependencyObject obj, KeyGesture? value) => obj.SetValue(ZoomResetKeyGestureProperty, value);
+
+        public static readonly DependencyProperty ZoomInKeyGestureProperty = DependencyProperty.RegisterAttached(
+            "ZoomInKeyGesture",
+            typeof(KeyGesture),
+            typeof(TextZoom),
+            new PropertyMetadata(OnKeyGestureChanged)
+        );
+
+        public static readonly DependencyProperty ZoomOutKeyGestureProperty = DependencyProperty.RegisterAttached(
+            "ZoomOutKeyGesture",
+            typeof(KeyGesture),
+            typeof(TextZoom),
+            new PropertyMetadata(OnKeyGestureChanged)
+        );
+
+        public static readonly DependencyProperty ZoomResetKeyGestureProperty = DependencyProperty.RegisterAttached(
+            "ZoomResetKeyGesture",
+            typeof(KeyGesture),
+            typeof(TextZoom),
+            new PropertyMetadata(OnKeyGestureChanged)
+        );
+
+
+        private static void OnKeyGestureChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            if (d is not FrameworkElement element)
+            {
+                return;
+            }
+
+            element.PreviewKeyDown -= OnElementPreviewKeyDown;
+            element.PreviewKeyDown += OnElementPreviewKeyDown;
+        }
+
+        private static void OnElementPreviewKeyDown(object sender, KeyEventArgs e)
+        {
+            if (sender is not FrameworkElement element)
+            {
+                return;
+            }
+
+            if (SetIfTriggered(GetZoomInKeyGesture(element),
+                s => s.ScaleX <= 3,
+                s =>
+                {
+                    s.ScaleX += Scale;
+                    s.ScaleY += Scale;
+                }))
+            {
+                return;
+            }
+
+            if (SetIfTriggered(GetZoomOutKeyGesture(element),
+                s => s.ScaleX >= .5,
+                s =>
+                {
+                    s.ScaleX -= Scale;
+                    s.ScaleY -= Scale;
+                }))
+            {
+                return;
+            }
+
+            SetIfTriggered(GetZoomResetKeyGesture(element),
+                s => true,
+                s =>
+                {
+                    s.ScaleX = 1;
+                    s.ScaleY = 1;
+                });
+
+
+            bool SetIfTriggered(KeyGesture? gesture, Predicate<ScaleTransform> predicate, Action<ScaleTransform> setter)
```

Comment:
I don't really see the benefit of this method. It seems the calls to this method above could be inlined as a switch expression to replace the `if (predicate(scaleTransform))` below?
```cs
double updatedScale = scaleTransform.ScaleX switch
{
    <= 3 => scaleTransform.ScaleX + Scale,
    >= .5 => scaleTransform.ScaleX - Scale,
    _ => 1
};
scaleTransform.ScaleX = updatedScale;
scaleTransform.ScaleY = updatedScale;
```

Reply:
OK, I reread the code to understand what was happening. To me using a consolidated method for all keybindings is not necessary, for the same reason we don't use one method to handle all command bindings. What do you think of breaking it up into split methods instead like this?
To me this is easier to maintain and easier to read, even if we get code that does somewhat the same thing (like the change callbacks)
```cs
using System.Windows;
using System.Windows.Input;
using System.Windows.Media;

namespace GitOut.Features.Wpf.AttachedProperties
{
    public class TextZoom
    {
        private static readonly double Scale = .2;

        public static KeyGesture? GetZoomInKeyGesture(DependencyObject obj) => (KeyGesture?)obj.GetValue(ZoomInKeyGestureProperty);
        public static KeyGesture? GetZoomOutKeyGesture(DependencyObject obj) => (KeyGesture?)obj.GetValue(ZoomOutKeyGestureProperty);
        public static KeyGesture? GetZoomResetKeyGesture(DependencyObject obj) => (KeyGesture?)obj.GetValue(ZoomResetKeyGestureProperty);
        public static void SetZoomInKeyGesture(DependencyObject obj, KeyGesture? value) => obj.SetValue(ZoomInKeyGestureProperty, value);
        public static void SetZoomOutKeyGesture(DependencyObject obj, KeyGesture? value) => obj.SetValue(ZoomOutKeyGestureProperty, value);
        public static void SetZoomResetKeyGesture(DependencyObject obj, KeyGesture? value) => obj.SetValue(ZoomResetKeyGestureProperty, value);

        public static readonly DependencyProperty ZoomInKeyGestureProperty = DependencyProperty.RegisterAttached(
            "ZoomInKeyGesture",
            typeof(KeyGesture),
            typeof(TextZoom),
            new PropertyMetadata(OnZoomInKeyGestureChanged)
        );

        public static readonly DependencyProperty ZoomOutKeyGestureProperty = DependencyProperty.RegisterAttached(
            "ZoomOutKeyGesture",
            typeof(KeyGesture),
            typeof(TextZoom),
            new PropertyMetadata(OnZoomOutKeyGestureChanged)
        );

        public static readonly DependencyProperty ZoomResetKeyGestureProperty = DependencyProperty.RegisterAttached(
            "ZoomResetKeyGesture",
            typeof(KeyGesture),
            typeof(TextZoom),
            new PropertyMetadata(OnResetZoomKeyGestureChanged)
        );

        private static void OnZoomInKeyGestureChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
        {
            if (d is FrameworkElement element)
            {
                element.PreviewKeyDown -= OnZoomIn;
                element.PreviewKeyDown += OnZoomIn;
            }
        }

        private static void OnZoomOutKeyGestureChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
        {
            if (d is FrameworkElement element)
            {
                element.PreviewKeyDown -= OnZoomOut;
                element.PreviewKeyDown += OnZoomOut;
            }
        }

        private static void OnResetZoomKeyGestureChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
        {
            if (d is FrameworkElement element)
            {
                element.PreviewKeyDown -= OnResetZoom;
                element.PreviewKeyDown += OnResetZoom;
            }
        }

        private static void OnZoomIn(object sender, KeyEventArgs e)
        {
            if (sender is FrameworkElement element
                && GetZoomInKeyGesture(element) is KeyGesture zoomIn
                && zoomIn is not null
                && zoomIn.Key == e.Key
                && zoomIn.Modifiers == Keyboard.Modifiers)
            {
                ScaleTransform transform = GetElementScaleTransform(element);
                if (transform.ScaleX <= 3)
                {
                    transform.ScaleX += Scale;
                    transform.ScaleY += Scale;
                }
            }
        }

        private static void OnZoomOut(object sender, KeyEventArgs e)
        {
            if (sender is FrameworkElement element
                && GetZoomOutKeyGesture(element) is KeyGesture zoomOut
                && zoomOut is not null
                && zoomOut.Key == e.Key
                && zoomOut.Modifiers == Keyboard.Modifiers)
            {
                ScaleTransform transform = GetElementScaleTransform(element);
                if (transform.ScaleX >= .5)
                {
                    transform.ScaleX -= Scale;
                    transform.ScaleY -= Scale;
                }
            }
        }

        private static void OnResetZoom(object sender, KeyEventArgs e)
        {
            if (sender is FrameworkElement element
                && GetZoomResetKeyGesture(element) is KeyGesture zoomReset
                && zoomReset is not null
                && zoomReset.Key == e.Key
                && zoomReset.Modifiers == Keyboard.Modifiers)
            {
                ScaleTransform transform = GetElementScaleTransform(element);
                transform.ScaleX = 1;
                transform.ScaleY = 1;
            }
        }

        private static ScaleTransform GetElementScaleTransform(FrameworkElement element) => (element.LayoutTransform is ScaleTransform scaleTransform)
            ? scaleTransform
            : (ScaleTransform)(element.LayoutTransform = new ScaleTransform());
    }
}
```